### DISCUSSION TITLE
WidgetKit & AIR

### DISCUSSION THREAD

**Discussion by fix89** [airsdk/Adobe-Runtime-Support#1740] [Ideas]
Hello, will a tool for creating widgets in iOS ever be developed?

**Comment by marchbold** [comment]
I doubt it will be possible to develop a widget using AIR, however you should be able to add a widget to your AIR application using the normal native development tools in Xcode and then packaging the widget in your AIR application. 

We have done similar things with other "iOS extensions" in the past.

**Comment by fix89** [comment]
Thank you for your replay! Is there a guide to do this (packaging the widget in AIR application).
Thanks!

**Reply by marchbold** [reply]
I haven't done a widget, but i have a much simpler "ios extension" guide that's used for sharing: 

https://docs.airnativeextensions.com/docs/share/appextension---introduction

I believe you should be able to pull out the key bits from this and just create the widget extension. Good luck :)
